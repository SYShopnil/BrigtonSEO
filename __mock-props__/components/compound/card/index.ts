import { eventHistoryCardData } from './event-history';
import { testimonialData, testimonialWithoutImageData } from './testimonial';
import {
  speakerCardNormalData,
  speakerCardTransparentData,
  teamData,
} from './person';
import { courseCardData } from './course';
import { upcomingCardData } from './upcoming';
import { pricingCardLargeData, pricingCardSmallData } from './pricing-card';
import { buyTicketCardData } from './buy-ticket';

export {
  testimonialData,
  pricingCardLargeData,
  pricingCardSmallData,
  courseCardData,
  upcomingCardData,
  speakerCardNormalData,
  speakerCardTransparentData,
  teamData,
  eventHistoryCardData,
  testimonialWithoutImageData,
  buyTicketCardData,
};
