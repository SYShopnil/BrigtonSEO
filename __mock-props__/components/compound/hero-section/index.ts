import {
  heroGeneralTrainingInCompany,
  heroGeneralTrainingScholarship,
  heroGeneralAboutCovid,
  heroGeneralAboutCreche,
  heroGeneralAboutDates,
  heroGeneralAboutFaq,
  heroGeneralAboutTeam,
  heroGeneralSpeakers,
  heroGeneralHostFringe,
  heroGeneralPrivacy,
  heroGeneralSchedule,
} from './general';
import {
  heroHomeData,
  heroHomeFringeEventsData,
  heroHomeTrainingData,
} from './home';

export {
  heroGeneralTrainingInCompany,
  heroGeneralTrainingScholarship,
  heroGeneralAboutCovid,
  heroGeneralAboutCreche,
  heroGeneralAboutDates,
  heroGeneralAboutFaq,
  heroGeneralAboutTeam,
  heroGeneralSpeakers,
  heroGeneralHostFringe,
  heroGeneralPrivacy,
  heroGeneralSchedule,
  heroHomeData,
  heroHomeFringeEventsData,
  heroHomeTrainingData,
};
