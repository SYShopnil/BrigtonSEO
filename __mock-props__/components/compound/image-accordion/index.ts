import {
  IImageAccordion,
  IImageAccordionCard,
  SectionThemeEnum,
} from '@src/types/compound';

export const imageAccordionCardData: IImageAccordionCard = {
  image:
    'https://res.cloudinary.com/dktw59lc2/image/upload/v1670331012/BrightonSEO/image-accordion/home/brightonseo_d4nh2j.png',
  title: 'BrightonSEO',
  link: 'https://www.google.com/',
  description:
    'Amet minim mollit non deserunt ullamco est sit aliqua dolor do amet sint. Velit officia consequat duis.',
};

export const imageAccordionCardsData: IImageAccordionCard[] = [
  { ...imageAccordionCardData },
  {
    ...imageAccordionCardData,
    title: 'Trainers',
    image:
      'https://res.cloudinary.com/dktw59lc2/image/upload/v1670331013/BrightonSEO/image-accordion/home/trainers_udeetj.png',
  },
  {
    ...imageAccordionCardData,
    title: 'Training Sessions',
    image:
      'https://res.cloudinary.com/dktw59lc2/image/upload/v1670331013/BrightonSEO/image-accordion/home/training-sessions_deaidv.png',
  },
  {
    ...imageAccordionCardData,
    title: 'London Training',
    image:
      'https://res.cloudinary.com/dktw59lc2/image/upload/v1670331012/BrightonSEO/image-accordion/home/london-training_ebqrhk.png',
  },
  {
    ...imageAccordionCardData,
    title: 'In-Company Training',
    image:
      'https://res.cloudinary.com/dktw59lc2/image/upload/v1670331012/BrightonSEO/image-accordion/home/in-company-training_ysgg2q.png',
  },
];

export const imageAccordionData: IImageAccordion = {
  sectionTheme: SectionThemeEnum.Blue,
  sectionHeading: {
    subTitle: 'About the events',
    title: 'Welcome to the Greatest SEO Conference 2022',
  },
  cards: imageAccordionCardsData,
};
