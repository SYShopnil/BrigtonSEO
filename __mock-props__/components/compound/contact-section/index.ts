/**
 * Internal Imports
 */
import { IContactSection } from '@src/types/compound';
import { formData } from '@root/__mock-props__/components/root/form';

// contactSectionWithoutAddressData
export const contactSectionWithoutAddressData: IContactSection = {
  sectionHeading: {
    title: 'Host For Fringe Events',
    description:
      'Amet minim mollit non deserunt ullamco est sit aliqua dolor do amet sint. Velit officia consequat duis enim velit mollit.',
  },
  contact: {
    phone: {
      label: 'Phone',
      value: '+456 9876 5432',
    },
  },
  formData: formData,
};

// contactSectionWithAddressData
export const contactSectionWithAddressData: IContactSection = {
  sectionHeading: {
    subTitle: 'Contact Us',
    title: 'We Here to help',
  },
  contact: {
    address: {
      label: 'Office',
      value:
        'Plus X Innovation Hub, c/o SiteVisibility Lewes Road, Brighton, BN2 4GL',
    },
    phone: {
      label: 'Phone',
      value: '+456 9876 5432',
    },
  },
  formData: formData,
};
