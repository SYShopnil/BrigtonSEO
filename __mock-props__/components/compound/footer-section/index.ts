import { IFooterSection } from '@src/types/compound';
import { IconEnum } from '@src/types/root';

export const footerSectionData: IFooterSection = {
  headingInfo: {
    title: 'Want something extra?',
    subTitle: 'Subscribe to newsletter',
    buttonText: 'Subscribe',
    buttonLink: '/subscribe',
    linkOpenNewTab: false,
  },
  contactInfo: {
    address:
      'c/o SiteVisibility, Plus X Innovation Hub, Lewes Road Brighton, BN2 4GL - 01273 805525',
    phoneNumberOne: '+965 6554 6598',
    phoneNumberTwo: '+965 6554 6599',
    email: 'brightonseo@gmail.com',
  },
  columnOne: {
    title: 'Company',
    linkFields: [
      {
        label: 'About Us',
        url: '/about-us',
      },
      {
        label: 'Tickets',
        url: '/tickets',
      },
      {
        label: 'Speakers',
        url: '/speakers',
      },
      {
        label: 'Training',
        url: '/training',
      },
      {
        label: 'Sponsors',
        url: '/sponsors',
      },
      {
        label: 'Career',
        url: '/career',
      },
      {
        label: 'Contact',
        url: '/contact',
      },
    ],
  },
  columnTwo: {
    title: 'Support',
    linkFields: [
      {
        label: 'FAQs',
        url: '/faqs',
      },
      {
        label: 'Code of Conduct',
        url: '/code-of-conduct',
      },
      {
        label: 'Privacy Policy',
        url: '/privacy-policy',
      },
      {
        label: 'Terms & Conditions',
        url: '/terms-and-conditions',
      },
      {
        label: 'Unsubscribe',
        url: '/unsubscribe',
      },
    ],
  },
  socialColumn: {
    title: 'Follow Us',
    socialLinks: [
      {
        platform: IconEnum.Facebook,
        url: 'https://www.facebook.com',
      },
      {
        platform: IconEnum.Twitter,
        url: 'https://www.twitter.com',
      },
      {
        platform: IconEnum.Instagram,
        url: 'https://www.instagram.com',
      },
      {
        platform: IconEnum.Youtube,
        url: 'https://www.youtube.com',
      },
    ],
  },
};
