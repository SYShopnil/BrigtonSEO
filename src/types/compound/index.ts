import { IContactSection, IContact } from './contact-section';
import {
  IListItem,
  IListContentSection,
  IContentHead,
  ListStyleEnum,
  IListContentCtaButton,
} from './list-content-section';
import { ITestimonialSection } from './testimonial-section';
import { IAuthor } from './common';
import {
  IHeroGeneral,
  HeroBgColorEnum,
  IHeroHome,
  EventInfoThemeEnum,
  IHeroLogo,
  IInfoCount,
} from './hero-section';
import { ITestimonial } from './card/testimonial';
import {
  IPricingCard,
  PriceBoxColorSchemeEnum,
  PriceBoxSizeEnum,
} from './card/pricing';
import { IPerson, ITeam, IPersonTransparent } from './card/person';
import { IFooterSection } from './footer-section';
import { IBuyTicketCard } from './card/buy-ticket';
import { IEventHistoryCard } from './card/event-history';
import { ICourseCard, ITag, TagBackgroundEnum } from './card/course';
import type { IUpcomingCard } from './card/upcoming';
import { UpcomingCardThemeEnum } from './card/upcoming';
import { IEventDate, IEventInfo, UserIconSizeEnum } from './common';
import { SectionThemeEnum } from './common';
import { ITextAccordion, ITextAccordionList, IFAQ } from './textAccordion';
import { IImageAccordion, IImageAccordionCard } from './image-accordion';

export type {
  IContactSection,
  IContact,
  IListItem,
  IListContentSection,
  IContentHead,
  ITestimonial,
  IBuyTicketCard,
  ICourseCard,
  ITag,
  IHeroGeneral,
  IPerson,
  IUpcomingCard,
  IFooterSection,
  IListContentCtaButton,
  IHeroHome,
  IEventInfo,
  IHeroLogo,
  IInfoCount,
  IPricingCard,
  ITextAccordion,
  ITextAccordionList,
  IFAQ,
  ITeam,
  IPersonTransparent,
  IEventHistoryCard,
  ITestimonialSection,
  IAuthor,
  IEventDate,
  IImageAccordion,
  IImageAccordionCard,
};

export {
  HeroBgColorEnum,
  UpcomingCardThemeEnum,
  ListStyleEnum,
  EventInfoThemeEnum,
  PriceBoxColorSchemeEnum,
  PriceBoxSizeEnum,
  TagBackgroundEnum,
  UserIconSizeEnum,
  SectionThemeEnum,
};
