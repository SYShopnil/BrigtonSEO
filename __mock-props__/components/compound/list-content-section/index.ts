import {
  IListItem,
  IListContentSection,
  ListStyleEnum,
  SectionThemeEnum,
} from '@src/types/compound';
import { IconEnum } from '@src/types/root';

export const protableText1 = [
  {
    _type: 'block',
    markDefs: [],
    style: 'normal',
    children: [
      {
        _type: 'span',
        marks: [],
        text: 'Personal data means information capable of identifying an individual. It does not include anonymised data.',
      },
    ],
  },
  {
    _type: 'block',
    markDefs: [],
    style: 'normal',
    children: [
      {
        _type: 'span',
        marks: [],
        text: 'We may process the following categories of personal data about you:',
      },
    ],
  },
];
export const protableText2 = [
  {
    _type: 'block',
    markDefs: [],
    style: 'normal',
    children: [
      {
        _type: 'span',
        marks: [],
        text: 'We know that the cost and logistics of organising childcare can prevent some people from taking part in personal development and networking opportunities like attending conferences.',
      },
    ],
  },
];

export const protableText3 = [
  {
    _type: 'block',
    markDefs: [],
    style: 'normal',
    children: [
      {
        _type: 'span',
        marks: [],
        text: `We want to alleviate some of that by offering an on-site creche at the Brighton Centre with experienced DBS checked and Paediatric First Aid trained staff.`,
      },
    ],
  },
  {
    _type: 'block',
    markDefs: [],
    style: 'normal',
    children: [
      {
        _type: 'span',
        marks: [],
        text: `They provide a wide section of activities including arts, crafts, music and movement, games, party games, snacks, free play, story time, and role play. We charge a nominal fee of Â£5 to secure a spot so we can accurately gauge demand and schedule adequate cover.`,
      },
    ],
  },
];
export const protableText4 = [
  {
    _type: 'block',
    markDefs: [],
    style: 'normal',
    children: [
      {
        _type: 'span',
        marks: [],
        text: `That includes any communication that you send to us whether that be through the contact form on our website, through email, text, social media messaging, social media posting or any other communication.`,
      },
    ],
  },
  {
    _type: 'block',
    markDefs: [],
    style: 'normal',
    children: [
      {
        _type: 'span',
        marks: [],
        text: `We process this data for the purposes of communicating with you, for record keeping and for the establishment, pursuance or defence of legal claims. Our lawful ground for this processing is our legitimate interests which in this case are to reply to communications sent to us, to keep records and to establish, pursue or defend legal claims.`,
      },
    ],
  },
];
export const protableText5 = [
  {
    _key: '3a81a8509ba1',
    _type: 'block',
    children: [
      {
        _key: '51a67de85a100',
        _type: 'span',
        marks: [],
        text: 'Contrary to popular belief,',
      },
      {
        _key: 'b85ddf6033d2',
        _type: 'span',
        marks: ['strong'],
        text: ' ',
      },
      {
        _key: '4ee7627a3c5d',
        _type: 'span',
        marks: ['strong'],
        text: 'Lorem Ipsum',
      },
      {
        _key: 'd3f9afaa052b',
        _type: 'span',
        marks: [],
        text: ' is not simply ',
      },
      {
        _key: 'ec13bf8f865b',
        _type: 'span',
        marks: ['em', 'strong', 'underline', '1bd567ab6e7f'],
        text: 'random text',
      },
      {
        _key: '63819c56e2aa',
        _type: 'span',
        marks: [],
        text: '. It has roots in a piece of classical Latin literature from 45 BC, making it over 2000 years old. ',
      },
    ],
    markDefs: [
      {
        _key: '1bd567ab6e7f',
        _type: 'link',
        href: 'https://www.google.com/',
      },
    ],
    style: 'normal',
  },
  {
    _key: 'e2bd2efe2273',
    _type: 'block',
    children: [
      {
        _key: 'a2d21495293b',
        _type: 'span',
        marks: [],
        text: 'Richard McClintock, a Latin professor at Hampden-Sydney College in Virginia, looked up one of the more obscure Latin words, consectetur, from a Lorem ',
      },
    ],
    markDefs: [],
    style: 'normal',
  },
];

const conentListDataWithIcon: IListItem[] = [
  {
    title: `Communication Data`,
    description: protableText4,
    icon: IconEnum.TrainFill,
  },
  {
    title: `Customer Data`,
    description: protableText5,
    icon: IconEnum.FlightTakeoffFill,
  },
  {
    title: `User Data`,
    description: protableText4,
    icon: IconEnum.BusFill,
  },
  {
    title: `Technical Data`,
    description: protableText5,
    icon: IconEnum.MotorbikeLine,
  },
];

const conentListData: IListItem[] = [
  {
    title: `Communication Data`,
    description: protableText4,
  },
  {
    title: `Customer Data`,
    description: protableText5,
  },
  {
    title: `User Data`,
    description: protableText4,
  },
  {
    title: `Technical Data`,
    description: protableText5,
  },
];

export const listConentBlockAboutBallot: IListContentSection = {
  sectionTheme: SectionThemeEnum.Blue,
  isDefaultBg: false,
  contentHeading: {
    sectionHeading: {
      title: 'Why a ballot?',
    },
    description: protableText2,
  },
  contentList: conentListData,
  listStyle: ListStyleEnum.Number,
};

export const listConentBlockAboutVenueTravel: IListContentSection = {
  sectionTheme: SectionThemeEnum.Blue,
  isDefaultBg: false,
  contentHeading: {
    sectionHeading: {
      title: 'Transportation is Near you ',
    },
    description: protableText2,
  },
  contentList: conentListDataWithIcon,
  listStyle: ListStyleEnum.Icon,
};

export const listConentBlockAboutCreche: IListContentSection = {
  sectionTheme: SectionThemeEnum.Blue,
  isDefaultBg: true,
  contentHeading: {
    sectionHeading: {
      subTitle: 'Write Text Here',
      title: 'BrightonSEO Crehe',
    },
    description: protableText2,
  },
  contentList: conentListData,
  listStyle: ListStyleEnum.None,
  button: {
    buttonText: 'Booking Now',
    linkOpenNewTab: true,
    link: 'https://www.google.com',
  },
};
