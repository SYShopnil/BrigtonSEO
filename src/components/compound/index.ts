import { ContactSection } from './contact-section';
import { ListContentSection } from './list-content-section';
import BuyTicketCard from './card/buy-ticket';
import { HeroGeneral, HeroHome } from './hero-section';
import CourseCard from './card/course';
import PricingCard from './card/pricing';
import { Testimonial } from './card/testimonial';
import Person from './card/person';
import UpcomingCard from './card/upcoming';
import FooterSection from './footer-section';
import { TestimonialSection } from './testimonial-section';
import { EventHistoryCard } from './card/event-history';
import { TextAccordion } from './textAccordion';
import { ImageAccordion } from './image-accordion';

export {
  ContactSection,
  CourseCard,
  FooterSection,
  ListContentSection,
  Testimonial,
  HeroGeneral,
  BuyTicketCard,
  PricingCard,
  Person,
  EventHistoryCard,
  UpcomingCard,
  TestimonialSection,
  HeroHome,
  TextAccordion,
  ImageAccordion,
};
