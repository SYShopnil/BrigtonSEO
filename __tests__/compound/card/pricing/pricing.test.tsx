import { render } from '@testing-library/react';
import { PricingCard } from '@src/components/compound';
import {
  pricingCardLargeData,
  pricingCardSmallData,
} from '@root/__mock-props__/components/compound/card';

/**
 * Test if the pricing card  is rendering correctly
 */
describe.each([pricingCardLargeData, pricingCardSmallData])(
  'Pricing Card',
  (props) => {
    it(`renders Pricing Card ${props.priceBoxSize} correctly`, () => {
      const { getByTestId } = render(<PricingCard {...props} />);
      const card = getByTestId('pricing-card');
      expect(card).toBeInTheDocument();
      expect(card).toMatchSnapshot();
    });
  }
);
