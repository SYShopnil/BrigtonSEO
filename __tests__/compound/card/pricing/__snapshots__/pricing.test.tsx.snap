// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pricing Card renders Pricing Card large correctly 1`] = `
<div
  class="relative overflow-hidden bg-white card radius-2xl shadow-One "
  data-testid="pricing-card"
>
  <div
    class="pricing-card secondary large text-center "
  >
    <div
      class="md:mb-6 mb-4"
    >
      <h3
        class="price-name capitalize"
      >
        basic
      </h3>
      <div
        class="description md:text-base text-sm capitalize pt-1"
      >
        in person
      </div>
    </div>
    <div
      class="price-divider opacity-50 h-[0.063rem] max-w-[10rem] md:mb-6 mb-4 mx-auto"
    />
    <div
      class="inline-flex"
    >
      <div>
        <h4
          class="price inline-flex items-center"
        >
          £
          270
          <span
            class="icon inline-block relative"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 12 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12 9H7v5H5V9H0V7h5V2h2v5h5v2z"
                fill-rule="evenodd"
              />
            </svg>
          </span>
        </h4>
      </div>
      <div
        class="vat-text inline-flex text-base capitalize items-center"
      >
        <span
          class="vat"
        >
          /
        </span>
        <div
          class="price-vat"
        >
          vat
        </div>
      </div>
    </div>
    <div
      class="md:mt-7 mt-5"
    >
      <a
        class="inline-flex"
        href="https://www.google.com"
        target="_self"
      >
        <span
          class="bs-button SkyblueBgTextWhite  btn-sm   group relative cursor-pointer overflow-hidden inline-flex gap-[1rem] items-center justify-center capitalize"
          data-testid="pricing-button"
        >
          <span>
            get ticket
          </span>
          <span
            class="absolute bs-button-bg inset-0"
          />
        </span>
      </a>
    </div>
  </div>
</div>
`;

exports[`Pricing Card renders Pricing Card small correctly 1`] = `
<div
  class="relative overflow-hidden bg-white card radius-2xl shadow-One "
  data-testid="pricing-card"
>
  <div
    class="pricing-card secondary small text-center "
  >
    <div
      class="md:mb-6 mb-4"
    >
      <h3
        class="price-name capitalize"
      >
        basic
      </h3>
    </div>
    <div
      class="inline-flex"
    >
      <div>
        <h4
          class="price inline-flex items-center"
        >
          £
          270
          <span
            class="icon inline-block relative"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 12 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12 9H7v5H5V9H0V7h5V2h2v5h5v2z"
                fill-rule="evenodd"
              />
            </svg>
          </span>
        </h4>
      </div>
      <div
        class="vat-text inline-flex text-base capitalize items-center"
      >
        <span
          class="vat"
        >
          /
        </span>
        <div
          class="price-vat"
        >
          vat
        </div>
      </div>
    </div>
    <div
      class="md:mt-7 mt-5"
    >
      <a
        class="inline-flex"
        href="https://www.google.com"
        target="_self"
      >
        <span
          class="bs-button SkyblueBgTextWhite  btn-ex-sm   group relative cursor-pointer overflow-hidden inline-flex gap-[1rem] items-center justify-center capitalize"
          data-testid="pricing-button"
        >
          <span>
            get ticket
          </span>
          <span
            class="absolute bs-button-bg inset-0"
          />
        </span>
      </a>
    </div>
  </div>
</div>
`;
